homeassistant/components/snooz/fan.py

```diff
@@ -0,0 +1,130 @@
+        self._attr_name = data.title
+        self._attr_unique_id = data.device.address
+        self._attr_supported_features = FanEntityFeature.SET_SPEED
+        self._attr_should_poll = False
+        self._is_on: bool | None = None
+        self._percentage: int | None = None
+
+    @callback
+    def _async_write_state_changed(self) -> None:
+        # cache state for restore entity
+        if not self.assumed_state:
+            self._is_on = self._device.state.on
+            self._percentage = self._device.state.volume
+
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Restore state and subscribe to device events."""
+        await super().async_added_to_hass()
+
+        if last_state := await self.async_get_last_state():
+            if last_state.state in (STATE_ON, STATE_OFF):
+                self._is_on = last_state.state == STATE_ON
+            else:
+                self._is_on = None
+            self._percentage = last_state.attributes.get(ATTR_PERCENTAGE)
+
+        self.async_on_remove(self._async_subscribe_to_device_change())
+
+    @callback
+    def _async_subscribe_to_device_change(self) -> Callable[[], None]:
+        return self._device.subscribe_to_state_change(self._async_write_state_changed)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect the device when removed."""
+        await self._device.async_disconnect()
+
+    @property
+    def percentage(self) -> int | None:
+        """Volume level of the device."""
+        return self._percentage if self.assumed_state else self._device.state.volume
+
+    @property
+    def is_on(self) -> bool | None:
+        """Power state of the device."""
+        return self._is_on if self.assumed_state else self._device.state.on
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return not self._device.is_connected or self._device.state is UnknownSnoozState
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn on the device."""
+        await self._async_execute_command(turn_on(percentage))
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the device."""
+        await self._async_execute_command(turn_off())
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the volume of the device."""
+        await self._async_execute_command(set_volume(percentage))
+
+    async def _async_execute_command(self, command: SnoozCommandData) -> None:
+        result = await self._device.async_execute_command(command)
+
+        if result.status == SnoozCommandResultStatus.CANCELLED:
+            _LOGGER.warning(
+                "Command %s was cancelled after %s", command, result.duration
+            )
```

Comment:
```
Oct 08 02:18:44 homeassistant homeassistant[530]: 2022-10-07 16:18:44.022 WARNING (MainThread) [homeassistant.components.snooz.fan] Command SetVolume(90%) was cancelled after 0:00:00.001020
Oct 08 02:18:44 homeassistant homeassistant[530]: 2022-10-07 16:18:44.029 WARNING (MainThread) [homeassistant.components.snooz.fan] Command SetVolume(96%) was cancelled after 0:00:00.000752
```

I see this in the log when making a lot of adjustments, but nothing is really wrong AFAICT so we should either not log or log at debug


```suggestion
            _LOGGER.debug(
                "Command %s was cancelled after %s", command, result.duration
            )
```